### PR TITLE
Adds muting history

### PR DESCRIPTION
**Pull request by dianabarsan**
# Description

For analytics purposes, a contact's `muting_history` log is saved in the sentinel info doc. Each time the muted status is changed, an entry is added in the `muting_history` log, following this format: 
```
{ 
  muted: [true|false],
  date: '<muting/unmuting sentinel date, in ISO8601 format>',
  report_id: '<report that triggered muting/unmuting>'
}
```

For analytics purposes, the contact's `muted` property now contains a human readable date (in ISO8601 format). When unmuting, the property is removed altogether, avoiding having mixed data types for the same property. 

The `muted` field value (when existent) is exposed to ContactViewModel, instead of a Boolean `true`. 

medic/medic-webapp#4767

# Review checklist

- [ ] Readable: Concise, well named, follows the [style guide](https://github.com/medic/medic-docs/blob/master/development/style-guide.md), documented if necessary.
- [ ] Documented: Announced in Changes.md in plain English. Configuration and user documentation on [medic-docs](https://github.com/medic/medic-docs/)
- [ ] Tested: Unit and/or e2e where appropriate
- [ ] Internationalised: All user facing text
- [ ] Backwards compatible: Works with existing data and configuration or includes a migration. Any breaking changes documented in Changes.md.

# License

The software is provided under AGPL-3.0. Contributions to this project are accepted under the same license.

### CODE REVIEW

**Comment by dianabarsan** [comment]
Could you please take a look, @garethbowen ? Thanks!

File: sentinel/src/lib/infodoc.js [hunk 1]
```diff
@@ -1,6 +1,7 @@
 const db = require('../db-pouch'),
   logger = require('../lib/logger'),
-  infoDocId = id => id + '-info';
+  infoDocId = id => id + '-info',
+  getDocId = infoDocId => infoDocId.slice(0, -5);
```
Comment (garethbowen) [inline sentinel/src/lib/infodoc.js]: I find it really confusing to put functions in the head part which is usually reserved for requirements (similar to imports in Java). What do you think? Is it clearer to have them split out with the rest of the functions?

File: sentinel/src/lib/muting_utils.js [hunk 2]
```diff
@@ -122,20 +164,21 @@ const updateMuteState = (contact, muted) => {
...
   });
 };
 
-const isMutedInLineage = doc => {
+const isMutedInLineage = (doc, returnId) => {
   let parent = doc && doc.parent;
   while (parent) {
     if (parent.muted) {
-      return true;
+      return returnId ? parent._id : true;
```
Comment (garethbowen) [inline sentinel/src/lib/muting_utils.js]: Why not always return the parent._id and rely on it being truthy?

Comment (dianabarsan) [reply sentinel/src/lib/muting_utils.js]: Done! I thought about it too.

File: sentinel/src/lib/infodoc.js (hunk 1 above)
Comment (dianabarsan) [reply sentinel/src/lib/infodoc.js]: Done!

**Comment by garethbowen** [comment]
@dianabarsan Good work. I'll leave it with you to merge to `master` and backport to `3.3.x` and `3.2.x`.

**Pull request merged**